src/routines.jl

```diff
@@ -47,6 +47,14 @@ end
     return N + (1 << (nbits - log2dim1(U))) * (length(U) - size(U, 2))
 end
 
+@inline function num_nonzero(nbits::Int, nctrls::Int, U::SDSparseMatrixCSC, N::Int = 1 << nbits)
```

Comment:
shouldn't this be in LuxurySparse?

Reply:
actually what is this function for? it seems not used in this PR?

Reply:
Actually, you are right, there are two interfaces can be removed.
This function should not be in LuxurySparse, because it is very related to control bits, and put blocks.
It is for determining the number of non-zero elements in the resulting matrix representation for put and control blocks.